### ISSUE TITLE
Get generator from instance 

### ISSUE THREAD

**Issue by dipterix** [r-lib/R6#286]
I wonder if it's possible to get generator class from its instance (like `R6::getR6Class(r6) -> generator`? Right now we can only get the class name.

**Comment by wch** [comment]
You're correct that you can only get the class name from the instance; there isn't a way to get the generator from an instance.

**Comment by dipterix** [comment]
Got it. Totally understand the potential issues.

Do you know if any class definitions can do this, S3 S4 R6 S7?
